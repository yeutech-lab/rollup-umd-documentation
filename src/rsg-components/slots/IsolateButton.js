import React from 'react';
import PropTypes from 'prop-types';
import MdFullscreen from 'react-icons/lib/md/fullscreen';
import MdFullscreenExit from 'react-icons/lib/md/fullscreen-exit';
import ToolbarButton from 'rsg-components/ToolbarButton';
import getUrl from '../../utils/getUrl';

const IsolateButton = ({ name, example, isolated }) =>
	isolated ? (
		<ToolbarButton href={getUrl({ anchor: true, slug: '/' })} title="Show all components">
			<MdFullscreenExit />
		</ToolbarButton>
	) : (
		<ToolbarButton href={getUrl({ name, example, isolated: true })} title="Open isolated">
			<MdFullscreen />
		</ToolbarButton>
	);

IsolateButton.propTypes = {
	name: PropTypes.string.isRequired,
	example: PropTypes.number,
	isolated: PropTypes.bool,
};

export default IsolateButton;
