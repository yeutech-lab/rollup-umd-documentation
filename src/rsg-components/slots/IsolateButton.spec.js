import React from 'react';
import IsolateButton from './IsolateButton';

it('should renderer a link to isolated mode', () => {
	const actual = shallow(<IsolateButton name="Pizza" />);

	expect(actual).toMatchSnapshot();
});

it('should renderer a link to example isolated mode', () => {
	const actual = shallow(<IsolateButton name="Pizza" example={3} />);

	expect(actual).toMatchSnapshot();
});

it('should renderer a link home in isolated mode', () => {
	const actual = shallow(<IsolateButton name="Pizza" isolated />);

	expect(actual).toMatchSnapshot();
});
