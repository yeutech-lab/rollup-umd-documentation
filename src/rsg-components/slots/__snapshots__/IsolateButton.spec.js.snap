// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renderer a link home in isolated mode 1`] = `
<Styled(ToolbarButton)
  href="blank#/"
  title="Show all components"
>
  <MdFullscreenExit />
</Styled(ToolbarButton)>
`;

exports[`should renderer a link to example isolated mode 1`] = `
<Styled(ToolbarButton)
  href="blank#!/Pizza/3"
  title="Open isolated"
>
  <MdFullscreen />
</Styled(ToolbarButton)>
`;

exports[`should renderer a link to isolated mode 1`] = `
<Styled(ToolbarButton)
  href="blank#!/Pizza"
  title="Open isolated"
>
  <MdFullscreen />
</Styled(ToolbarButton)>
`;
